src/nft/NFTFactory.ts

```diff
@@ -0,0 +1,90 @@
+import { isRequired } from '../utils/validations'
+import StorageType from './storage/StorageType'
+import { NFT_METADATA_FORMAT } from './types'
+import { NFT } from './NFT'
+import { NFT721Manager } from '../managers/NFT721Manager'
+import errors from '../types/errors'
+import { NftMetadata } from '../types/interfaces'
+
+export class NFTFactory {
+  protected _manager: NFT721Manager
+  protected _storage: StorageType
+  protected _nftFormat: NFT_METADATA_FORMAT
+
+  /**
+   * @param {NFT721Manager} nftManager - The manager of the contract.
+   * @param {StorageType} storage - A storage to bring the necessary data.
+   * @param {NFT_METADATA_FORMAT} nftFormat - The format of the nft metadata.
+   * @returns NFTFactory object.
+   */
+  constructor(
+    nftManager: NFT721Manager = isRequired('nftManager'),
+    storage: StorageType = isRequired('storage'),
+    nftFormat: NFT_METADATA_FORMAT = isRequired('nftFormat')
+  ) {
+    this._manager = nftManager
+    this._storage = storage
+    this._nftFormat = nftFormat
+  }
+
+  // getters
+  get manager(): NFT721Manager {
+    return this._manager
+  }
+
+  get storage(): StorageType {
+    return this._storage
+  }
+
+  get nftFormat(): NFT_METADATA_FORMAT {
+    return this._nftFormat
+  }
+  // end getters
+
+  /**
+   * Create a new NFT instance from a tokenId.
+   * @param  {string} tokenId - The token id of the nft.
+   * @return {Promise<NFT>} - Returns a new NFT instance.
+   */
+  async createNFT(tokenId: string = isRequired('tokenId')): Promise<NFT> {
+    if (!this._manager.implements('tokenURI', ['uint256'])) {
+      throw errors.TOKEN_URI_NOT_IMPLEMENTED(this._manager.contractAddr || '') // Avoid forbidden-non-null-assertion
+    }
+
+    let tokenUri = ''
+    try {
+      const { value } = await this._manager.execute({
+        method: 'tokenURI(uint256)',
+        params: [tokenId]
+      })
+      tokenUri = value
```

Comment:
I can't do that because in that case it doesn't assign the value to `let tokenUri = ''` but it lives only inside the try and they are taken as two independent variables

Reply:
yeah, that's true. I didn't take that into account when i did the review. 
Pain in js scopes (?